dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathEntry.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+/** An entry in Java class path. */
+interface ClassPathEntry {
```

Comment:
Should we be able to use this type exclusively? If we have to do any casting or reflection to determine the real type, the design is incomplete.

Reply:
Almost. ArtifactClassPathEntry will appear only when Linkage Checker needs to read Artifact's coordinates. It's intended, as that's the reason why I'm class path entries artifact-aware when class paths are from Maven artifacts.

Reply:
It sounds like this class hierarchy doesn't match the use case. We shouldn't need reflection or instanceof.

Reply:
Updated ClassPathEntry as a concrete class. ArtifactClassPathEntry is a spacial case of ClassPathEntry that has associated Maven artifact. It will use `instanceof` to see a class path entry has associated Maven artifact. I don't think this use of `instanceof` means a bad class hierarchy.

Reply:
Updated. ClassPathEntry now has `getArtifact`, which returns `null` by default. ArtifactClassPathEntry overrides it. Now we will not need to have `instanceof`.